test/extended/networking/tuning.go

```diff
@@ -43,6 +45,14 @@ var whitelistedSysctls = []SysctlVariant{
 	// {Sysctl: "net.ipv6.neigh.IFNAME.retrans_time_ms", Value: "1010", Path: "/proc/sys/net/ipv6/neigh/net1/retrans_time_ms"},
 }
 
+// getPodNodeName returns the name of the node the pod is scheduled on
+func getPodNodeName(client clientset.Interface, namespace, name string) string {
+	pod, err := client.CoreV1().Pods(namespace).Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
it's not enough to just immediately re-retrieve; you have to wait for the pod to get scheduled by kube-scheduler, which might take longer-than-it-takes-you-to-re-retrieve-once.

there ought to be a helper for this...

actually, now that I think about it, the usual trick is to pick a node beforehand with `e2enode.GetRandomReadySchedulableNode()` and then set `pod.Spec.NodeName` on the first pod too

Reply:
> it's not enough to just immediately re-retrieve; you have to wait for the pod to get scheduled by kube-scheduler, which might take longer-than-it-takes-you-to-re-retrieve-once.

@danwinship this is actually getting executed after CreateExecPodOrFail() which always waits for the pod to be running.